lib/buffer.js

```diff
@@ -141,7 +141,10 @@ function fromJsonObject(that, object) {
 
 function allocate(that, length) {
   var fromPool = length !== 0 && length <= Buffer.poolSize >>> 1;
-  that.parent = fromPool ? palloc(that, length) : alloc(that, length);
+  if (fromPool)
```

Comment:
would it be alright to explicitly assign `.parent = null` above here so we can keep the same object shape?